File: codes/3.ProgrammingLanguage/编程语言试水/Score100WordsAnalyzer.cs
using System.Text;

public class Score100WordsAnalyzer
{
    public static void Main(string[] args)
    {
        long start = DateTime.Now.Ticks;
        List<string> score100Words = File.ReadAllLines("Dictionary.txt", Encoding.UTF8)
            .Where(line => !line.Contains('-') && !line.Contains(' ') && !line.Contains('='))
            .Where(word => word.Sum(c => c - 'a' + 1) == 100)
            .ToList();
        Console.WriteLine(string.Join(",", score100Words));
        Console.WriteLine("\nwords number: " + score100Words.Count);
        Console.WriteLine("        time: " + (DateTime.Now.Ticks - start) / TimeSpan.TicksPerMillisecond + "ms");
    }
}
/*
 * 编码体验：
 * 1、方法名字母都是大写的
 * 2、IDE（CLion）中点击方法看详情时，看不到对应的文档，比如方法含义、参数含义、返回值含义等
 * 3、字符串的Sum函数在本例中很方便
 * 4、顶部引用的类比Java少一些
 * 5、左花括号要单独占一行
 */

File: codes/3.ProgrammingLanguage/编程语言试水/Score100WordsAnalyzer.jl
using Dates

start = Dates.now()
score100_words = []
for line in readlines("Dictionary.txt")
    if !contains(line, " ") && !contains(line, "-") && !contains(line, "=")
        score = sum([Int(c) - Int('a') + 1 for c in line])
        if score == 100
            push!(score100_words, line)
        end
    end
end

println(score100_words)
println("\nWords number:", length(score100_words))
println("        time:", Dates.now() - start)
